framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -869,7 +869,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
Do you mean all `unmodifiableSet` calls should be removed or we just leave one `unmodifiableSet` call in `getSupportedTypeQualifiers()`, and remove all other `unmodifiableSet` calls in `createSupportedTypeQualifiers()`?

I think latter is safer, because in `getSupportedTypeQualifiers()`, after calling `createSupportedTypeQualifiers()`, the `supportedQuals` should become unmodifiable.